ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,366 @@
+        SELECT distinct *
+          WHERE {
+          %SUBJECT% ?p ?o .
+          OPTIONAL {?o  rdfs:label ?label .}
+        }
+        """
+
+        annotationQuery = annotationQuery.replace("%SUBJECT%", subject)
+        # now = time.time()
+        results = self._rdfGraph.query(annotationQuery)
+        rows = [row.asdict() for row in results]
+        # responseTime = time.time()-now
+        # if responseTime > self.RESPONSETIME_LOGGING_THRESHOLD:
+        #     print('annotationQuery', responseTime)
+        #     print(annotationQuery)
+
+        for row in rows:
+            for k in row:
+                row[k] = row[k].toPython()
+            row['s'] = subject
+
+        locationQueryTemplate = """
+        PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+             PREFIX OBO: <http://purl.obolibrary.org/obo/>
+        SELECT distinct *
+        WHERE {
+        %SUBJECT%    a  OBO:SO_0001059  .
+        %SUBJECT%   ?p ?o .
+        OPTIONAL {?o  rdfs:label ?label .} .
+        }
+        """
+
+        locationRows = []
+        uniqueLocations = sets.Set()
+        for row in rows:
+            if row['p'] == 'http://purl.org/oban/association_has_subject':
+                uniqueLocations.add("<" + row['o'] + ">")
+
+        for location in uniqueLocations:
+            locationQuery = locationQueryTemplate.replace(
+                "%SUBJECT%", location)
+            results = self._rdfGraph.query(locationQuery)
+            locationRows = [locationRow.asdict() for locationRow in results]
+            for locationRow in locationRows:
+                for k in locationRow:
+                    locationRow[k] = locationRow[k].toPython()
+                locationRow['s'] = location
+            # if responseTime > self.RESPONSETIME_LOGGING_THRESHOLD:
+            #     print('locationQuery', responseTime)
+            #     print(locationQuery)
+
+        annotation = self.flatten(rows)
+        location = self.flatten(locationRows)
+        annotation['location'] = location
+        return annotation
+
+    def flatten(self, dict):
+        """
+        Given a dict of dicts,
+        flatten it to a single dict using predicate as keys
+        For multiple occurrences of a predicate, create an array
+        Each value in the dict is an object {val:'x', label:'y'}
+        The value of 's' (subject) is copied to the 'id' property
+        """
+        a = {}
+        for row in dict:
+            obj = {'val': row['o']}
+            if 'label' in row:
+                obj['label'] = row['label']
+
+            if row['p'] in a and \
+                    a[row['p']].__class__.__name__ != "list":
+                asIs = a[row['p']]
+                a[row['p']] = []
+                a[row['p']].append(asIs)
+
+            if row['p'] in a:
+                a[row['p']].append(obj)
+            else:
+                a[row['p']] = obj
+
+            a['id'] = row['s']
+        return a
+
+    def toGA4GH(self, annotation):
```

Comment:
It feels like we're munging a lot of stuff together in this class. Perhaps we could have another class called Annotation, which takes care of the process of converting one of these low-level annotations into a GA4GH object via the ``toProtocolElement`` method? This would make unit testing a lot easier and I think clarify the purpose of classes a bit more.